misk/testing/src/main/kotlin/misk/logging/RealLogCollector.kt

```diff
@@ -0,0 +1,62 @@
+package misk.logging
+
+import ch.qos.logback.classic.Level
+import ch.qos.logback.classic.Logger
+import ch.qos.logback.classic.spi.ILoggingEvent
+import ch.qos.logback.core.UnsynchronizedAppenderBase
+import com.google.common.util.concurrent.AbstractIdleService
+import com.google.common.util.concurrent.Service.State.NEW
+import org.slf4j.LoggerFactory
+import javax.inject.Singleton
+import kotlin.reflect.KClass
+
+@Singleton
+internal class RealLogCollector : AbstractIdleService(), LogCollector {
+  private val events = mutableListOf<ILoggingEvent>()
+
+  private val appender = object : UnsynchronizedAppenderBase<ILoggingEvent>() {
```

Comment:
why an unsynchronized appender, just to synchronize below?

Reply:
I’ve got to synchronize with both producers and consumers